src/inputs/pipe.c

```diff
@@ -302,6 +304,51 @@ parse_progress(struct input_metadata *m, char *progress)
   m->song_length = (end - start) * 10 / 441; // Convert to ms based on 44100
 }
 
+static void
+parse_volume(char *volume)
+{
+  char *value_begin;
+  char *value_next;
+  char *value_end;
+  float airplay_volume;
+
+  if (!(value_begin = strtok_r(volume, ",", &value_next)))
+    return;
+
+  airplay_volume = strtof(value_begin, &value_end);
+
+  if ((errno == ERANGE) || (value_begin == value_end))
+    {
+      char no_number[] = "First token is not a number.";
+      char *failure_detail = errno == ERANGE ? strerror(errno) : no_number;
+      DPRINTF(E_LOG, L_PLAYER, "Invalid airplay volume in string (%s): %s\n", volume, failure_detail);
+      return;
+    }
+
+  if (strcmp(value_next, pipe_metadata_pvol_enabled_suffix) != 0)
+    {
+      DPRINTF(E_DBG, L_PLAYER, "Not applying airplay volume while software volume control is enabled (%s)\n", volume);
+      return;
+    }
+
+  if (airplay_volume > -144.5 && airplay_volume < -143.5)
```

Comment:
I followed the suggestions that say never do "==" or "!=" on floats/doubles as cutoff or rounding effects may otherwise produce unpredictable effects (e.g. 144.00000001). In this case it may absolutely not be necessary but I wanted to be safe. A cast to "int" would do the same and makes code easier to read.